engine/src/main/java/org/pentaho/di/trans/steps/textfileinput/TextFileInput.java

```diff
@@ -0,0 +1,1658 @@
+@Deprecated
+public class TextFileInput extends BaseStep implements StepInterface {
+  private static Class<?> PKG = TextFileInputMeta.class; // for i18n purposes, needed by Translator2!!
+
+  private static final int BUFFER_SIZE_INPUT_STREAM = 500;
+
+  private TextFileInputMeta meta;
+
+  private TextFileInputData data;
+
+  private long lineNumberInFile;
+
+  public TextFileInput( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr, TransMeta transMeta,
+      Trans trans ) {
+    super( stepMeta, stepDataInterface, copyNr, transMeta, trans );
+  }
+
+  public static final String getLine( LogChannelInterface log, InputStreamReader reader, int formatNr,
+      StringBuilder line ) throws KettleFileException {
+    EncodingType type = EncodingType.guessEncodingType( reader.getEncoding() );
+    return getLine( log, reader, type, formatNr, line );
+  }
+
+  public static final String getLine( LogChannelInterface log, InputStreamReader reader, EncodingType encodingType,
+      int formatNr, StringBuilder line ) throws KettleFileException {
+    int c = 0;
+    line.setLength( 0 );
+    try {
+      switch ( formatNr ) {
+        case TextFileInputMeta.FILE_FORMAT_DOS:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isReturn( c ) || encodingType.isLinefeed( c ) ) {
+              c = reader.read(); // skip \n and \r
+              if ( !encodingType.isReturn( c ) && !encodingType.isLinefeed( c ) ) {
+                // make sure its really a linefeed or cariage return
+                // raise an error this is not a DOS file
+                // so we have pulled a character from the next line
+                throw new KettleFileException( BaseMessages.getString( PKG, "TextFileInput.Log.SingleLineFound" ) );
+              }
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_UNIX:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) || encodingType.isReturn( c ) ) {
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_MIXED:
+          // in mixed mode we suppose the LF is the last char and CR is ignored
+          // not for MAC OS 9 but works for Mac OS X. Mac OS 9 can use UNIX-Format
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) ) {
+              return line.toString();
+            } else if ( !encodingType.isReturn( c ) ) {
+              if ( c >= 0 ) {
+                line.append( (char) c );
+              }
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    } catch ( KettleFileException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)